label_studio/users/mixins.py

```diff
@@ -7,3 +10,20 @@ def has_permission(self, user):
         if user.active_organization in self.organizations.all():
             return True
         return False
+
+
+class UserRelatedManagerMixin:
+    @cached_property
+    def get_all_field_names(self):
+        return [field.name for field in self.model._meta.fields]
+
+    def get_queryset(self):
+        qs = super().get_queryset()
+        if 'is_deleted' in self.get_all_field_names:
+            qs = qs.filter(is_deleted=False)
+        else:
+            qs = qs.filter(user__is_deleted=False)
```

Comment:
Why is `get_all_field_names` needed? Will there be a user model that doesn't have the `is_deleted` field defined on it?

Reply:
Originally I had anticipated other models with relations to user would benefit from a manager so it was written to handle that.  Since then I determined that wouldn't really be necessarily.  I'll probably be removing this mixin entirely and baking it in to the current user manager.  However, looks the check to self.get_all_field_names is still needed, tests fail without it because this manager is also used in tests and in migrations so during test migration setup, the users model is called before is_deleted has been added.  I wonder if there's a more elegant way to do this 🤔

Reply:
Seems easier to just catch the exception and that way we don't need `get_all_field_names` on the filter of `is_deleted` so if its not there we just ignore the exception and carry on.